docs/apis/core/customfields/index.md

```diff
@@ -0,0 +1,168 @@
+
+How custom fields are used depends entirely on the situation. The following handler methods will help you to retrieve custom fields values for the given instance(s):
+
+```php
+$handler->export_instance_data()
+$handler->export_instance_data_object()
+$handler->display_custom_fields_data()
+```
+
+Additional methods for advanced usage:
+
+```php
+$handler->get_instance_data()
+$handler->get_instances_data()
+$handler->get_instance_data_for_backup()
+```
+
+Method restore_instance_data_from_backup() exists in the handler class but not implemented
+
+To retrieve the list of custom fields used in the given component/area/itemid you can use:
+
+```php
+$handler->get_categories_with_fields()
+$handler->get_fields()
+```
+
+List of fields is cached in the handler and these two functions can be called multiple times.
+
+#### Example code for course custom fields
+
+This function will return all the custom fields for a given courseid:
+
+```php
+function get_course_metadata($courseid) {
+    $handler = \core_customfield\handler::get_handler('core_course', 'course');
+    // This is equivalent to the line above.
+    //$handler = \core_course\customfield\course_handler::create();
+    $datas = $handler->get_instance_data($courseid);
+    $metadata = [];
+    foreach ($datas as $data) {
+        if (empty($data->get_value())) {
+            continue;
+        }
+        $cat = $data->get_field()->get_category()->get('name');
+        $metadata[$data->get_field()->get('shortname')] = $cat . ': ' . $data->get_value();
+    }
+    return $metadata;
+}
+```
+
+### Privacy API
+
+Custom fields API does not export or delete any data because it does not know how custom fields are used, what data is considered user data and if it is considered private or shared data.
+
+Plugins that store user information in custom fields should link subsystem in their get_metadata:
+
+```php
+$collection->link_subsystem('core_customfield', 'privacy:metadata:customfieldpurpose');
+```
+
+And they can use the following methods in the export/delete functions:
+
+```php
+use core_customfield\privacy\provider as customfield_provider;
+
+customfield_provider::get_customfields_data_contexts()
+customfield_provider::export_customfields_data()
+customfield_provider::delete_customfields_data()
+customfield_provider::delete_customfields_data_for_context()
+```
+
+In case when custom fields configuration is considered to be user data (configuration means the definition of the fields, not the instance data), there are also couple of methods to help with privacy API implementations:
+
+```php
+customfield_provider::get_customfields_configuration_contexts()
+customfield_provider::delete_customfields_configuration()
+customfield_provider::delete_customfields_configuration_for_context()
+```
+
+Export of configuration was not yet implemented at the time of writing this because of difficult implementation and very unclear use case. If it is needed please feel free to contribute to Moodle.
```

Comment:
```suggestion

:::info

Export of configuration was not yet implemented at the time of writing this because of difficult implementation and very unclear use case. If it is needed please feel free to contribute to Moodle.

:::
```